### PR TITLE
refactor: Global var instead of ref for current coordinates

### PR DESCRIPTION
**Pull request by gorandalum**
Using a global variable is easier for clients since they don't need
to go through the context. Also they don't need to provide the ref
in the dependency array of hooks, which the linter demands even
though it doesn't make a difference.

### CODE REVIEW

**Review by reidzeibel — APPROVED** [review]
I thought it's going to use `var` but it was a `const`! 🫨

File: src/GeolocationContext.tsx [hunk 1]
```diff
@@ -122,7 +131,6 @@ const defaultState: GeolocationState = {
   location: null,
   locationError: null,
   getCurrentCoordinates: () => Promise.resolve(undefined),
-  currentCoordinatesRef: undefined,
```
[the remark this answers is not in the document]
Comment (gorandalum) [reply src/GeolocationContext.tsx:125]: Not sure I follow 🤔 The suggestion is that the global variable is returned through the context state as well?

[the remark this answers is not in the document]
Comment (marius-at-atb) [reply src/GeolocationContext.tsx:125]: I mean can we remove
```ts
const currentCoordinatesRef = useRef<Coordinates | undefined>();
```
and 
```ts
useEffect(() => {
    currentCoordinatesRef.current = state.location?.coordinates;
  }, [state.location?.coordinates]);
```
and then just use 
```ts
if (currentCoordinatesGlobal) {
          resolve(currentCoordinatesGlobal);
```
inside `getCurrentCoordinates`?

[the remark this answers is not in the document]
Comment (gorandalum) [reply src/GeolocationContext.tsx:125]: Oh wow, I looked into this `getCurrentCoordinates` and I am kinda surprised there are no bugs connected to it. It is a reactive function which get a new signature on status change and language change, and it is used as a dependency in other `useEffect`s. Also it is not preferable to have a function which starts with `get` have a bunch of side effects. 

Well well, no time to look into this now, but I changed the logic using ref to instead use the global variable.

**Comment by tormoseng** [comment]
✅  Quick regression on places we use coordinates like my position/choose from map in travel search and departures (+ details), along with default chosen zones and more thorougly in the map.

**Pull request merged**